xline-client/src/clients/kv.rs

```diff
@@ -27,4 +31,762 @@ impl KvClient {
+            cmp,
+            CompareTarget::Value,
+            TargetUnion::Value(value.into()),
+        )
+    }
+
+    /// Compares the lease id of the given key.
+    #[inline]
+    pub fn lease(key: impl Into<Vec<u8>>, cmp: CompareResult, lease: i64) -> Self {
+        Self::new(key, cmp, CompareTarget::Lease, TargetUnion::Lease(lease))
+    }
+
+    /// Sets the comparison to scan the range [key, end).
+    #[inline]
+    #[must_use]
+    pub fn with_range(mut self, end: impl Into<Vec<u8>>) -> Self {
+        self.0.range_end = end.into();
+        self
+    }
+
+    /// Sets the comparison to scan all keys prefixed by the key.
+    #[inline]
+    #[must_use]
+    pub fn with_prefix(mut self) -> Self {
+        self.0.range_end = KeyRange::get_prefix(&self.0.key);
+        self
+    }
+}
+
+/// Transaction operation.
+#[derive(Debug, Clone)]
+pub struct TxnOp {
+    /// The inner txn op request
+    inner: xlineapi::Request,
+}
+
+impl TxnOp {
+    /// `Put` operation.
+    #[inline]
+    #[must_use]
+    pub fn put(reqeust: PutRequest) -> Self {
+        TxnOp {
+            inner: xlineapi::Request::RequestPut(reqeust.into()),
+        }
+    }
+
+    /// `Get` operation.
+    #[inline]
+    #[must_use]
+    pub fn range(request: RangeRequest) -> Self {
+        TxnOp {
+            inner: xlineapi::Request::RequestRange(request.into()),
+        }
+    }
+
+    /// `Delete` operation.
+    #[inline]
+    #[must_use]
+    pub fn delete(request: DeleteRangeRequest) -> Self {
+        TxnOp {
+            inner: xlineapi::Request::RequestDeleteRange(request.into()),
+        }
+    }
+
+    /// `Txn` operation.
+    #[inline]
+    #[must_use]
+    pub fn txn(txn: Txn) -> Self {
+        TxnOp {
+            inner: xlineapi::Request::RequestTxn(txn.into()),
+        }
+    }
+}
+
+impl From<TxnOp> for xlineapi::Request {
+    #[inline]
+    fn from(op: TxnOp) -> Self {
+        op.inner
+    }
+}
+
+/// Transaction of multiple operations.
+#[derive(Debug)]
+pub struct Txn {
+    /// the inner txn request
+    req: xlineapi::TxnRequest,
+    /// if `when` have be set
+    c_when: bool,
+    /// if `then` have be set
+    c_then: bool,
+    /// if `else` have be set
+    c_else: bool,
+}
+
+impl Txn {
+    /// Creates a new transaction.
+    #[inline]
+    #[must_use]
+    pub const fn new() -> Self {
+        Self {
+            req: xlineapi::TxnRequest {
+                compare: Vec::new(),
+                success: Vec::new(),
+                failure: Vec::new(),
+            },
+            c_when: false,
+            c_then: false,
+            c_else: false,
+        }
+    }
+
+    /// Takes a list of comparison. If all comparisons passed in succeed,
+    /// the operations passed into `and_then()` will be executed. Or the operations
+    /// passed into `or_else()` will be executed.
+    ///
+    /// # Panics
+    ///
+    /// panics if `when` is called twice or called after `when` or called after `or_else`
+    #[inline]
```

Comment:
I guess we could remove `#[inline]` here according to [this](https://matklad.github.io/2021/07/09/inline-in-rust.html#Inlining-in-Practice)
> Fifth, mind generic functions. It’s not too wrong to say that generic functions are implicitly inline. As a result, they often are a cause for code bloat. Generic functions, especially in libraries, should be written to minimize unwanted inlining. To give an example from [wat](https://github.com/bytecodealliance/wasm-tools/blob/0486fb4de505b8116a0034bdde4918cd783325b9/crates/wat/src/lib.rs#L214-L222):

Reply:
The clippy requires inline in public functions. As mentioned in the article, the generic functions are implicitly inlined, adding `#[inline]` just explicitly does it. I think what the author means is to add method delegation in generic functions, but that sounds like premature optimization?

Reply:
OK, if clippy doesn't allow it, then just keep it.